private/src/avb_streamhandler_app_socket/server.cpp

```diff
@@ -0,0 +1,1750 @@
+  responseSocketIpcStruct.result = IasAvbResult::eIasAvbResultNotImplemented;
+
+  return responseSocketIpcStruct;
+}
+
+/// CreateTransmitAvbVideoStream command
+AvbStreamHandlerSocketIpc::responseSocketIpc CreateTransmitAvbVideoStream::execute (
+			IasMediaTransportAvb::IasAvbStreamHandler *avbStreamHandler,
+			AvbStreamHandlerSocketIpc::requestSocketIpc *requestedCmdStruct)
+{
+  IasMediaTransportAvb::IasAvbResult resultx = IasAvbResult::eIasAvbResultErr;
+
+  if (getVideoFormat(requestedCmdStruct->format) == -1)
+  {
+    requestedCmdStruct->format=1; // default eIasAvbVideoFormatRtp
+  }
+
+  resultx = avbStreamHandler->createTransmitVideoStream (
+          requestedCmdStruct->srClass,
+          requestedCmdStruct->maxPacketRate,
+          requestedCmdStruct->maxPacketSize,
+          getVideoFormat(requestedCmdStruct->format),
+          requestedCmdStruct->clockId,
+          getAssignMode(requestedCmdStruct->assignMode),
+          requestedCmdStruct->networkStreamId,
+          requestedCmdStruct->dmac,
+          requestedCmdStruct->active
+          );
+
+  std::cout << "\tResult: " << getResultString(resultx) << std::endl;
+  AvbStreamHandlerSocketIpc::responseSocketIpc responseSocketIpcStruct;
+  responseSocketIpcStruct.command = requestedCmdStruct->command;
+  responseSocketIpcStruct.result = getResultString(resultx);
+
+  //streamid may be updated by avbstreamhandler if it is not available
+  responseSocketIpcStruct.oStreamId = requestedCmdStruct->networkStreamId;
+
+  return responseSocketIpcStruct;
+}
+
+/// CreateReceiveVideoStream command
+AvbStreamHandlerSocketIpc::responseSocketIpc CreateReceiveVideoStream::execute (
+			IasMediaTransportAvb::IasAvbStreamHandler *avbStreamHandler,
+			AvbStreamHandlerSocketIpc::requestSocketIpc *requestedCmdStruct)
+{
+  IasMediaTransportAvb::IasAvbResult resultx = IasAvbResult::eIasAvbResultErr;
+
+  if (getVideoFormat(requestedCmdStruct->format) == -1)
+  {
+    requestedCmdStruct->format=1; // default eIasAvbVideoFormatRtp
+  }
+
+  resultx = avbStreamHandler->createReceiveVideoStream (
+          requestedCmdStruct->srClass,
+          requestedCmdStruct->maxPacketRate,
+          requestedCmdStruct->maxPacketSize,
+          getVideoFormat(requestedCmdStruct->format),
+          requestedCmdStruct->networkStreamId,
+          requestedCmdStruct->dmac
+          );
+
+  std::cout << "\tResult: " << getResultString(resultx) << std::endl;
+  AvbStreamHandlerSocketIpc::responseSocketIpc responseSocketIpcStruct;
+  responseSocketIpcStruct.command = requestedCmdStruct->command;
+  responseSocketIpcStruct.result = getResultString(resultx);
+
+  //streamid may be updated by avbstreamhandler if it is not available
+  responseSocketIpcStruct.oStreamId = requestedCmdStruct->networkStreamId;
+
+  return responseSocketIpcStruct;
+}
+
+/// CreateLocalVideoStream command
+// TODO: Not implemented/enabled now. To update after vstreaming implementation is ready
```

Comment:
Oh, I guess that this does not work. One needs to properly implement it. Check if it is a requirement before putting any efforts, please.

Reply:
will do. for now we will submit as is.